peer_test.go

```diff
@@ -204,6 +204,27 @@ func TestPublicKeyExtraction(t *testing.T) {
 	}
 }
 
+func TestValidate(t *testing.T) {
+	// Empty peer ID invalidates
+	err := ID("").Validate()
+	if err == nil {
+		t.Error("expected error")
+	} else if err.Error() != "empty peer ID" {
+		t.Error("expected error message: empty peer ID")
+	}
+
+	// Non-empty peer ID validates
+	p, err := tu.RandPeerID()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = p.Validate()
+	if err != nil {
+		t.Error("expected nil, but found " + err.Error())
+	}
+}
+
```

Comment:
I Have used t.Error() for function that is being tested here. For failure of any other functions t.Fatal() has been used. Kinda inspired from other tests here,